streams/src/main/java/org/apache/kafka/streams/kstream/Produced.java

```diff
@@ -80,7 +81,7 @@ protected Produced(final Produced<K, V> produced) {
      */
     public static <K, V> Produced<K, V> with(final Serde<K> keySerde,
                                              final Serde<V> valueSerde,
-                                             final StreamPartitioner<? super K, ? super V> partitioner) {
+                                             final StreamPartitioner<K, V> partitioner) {
```

Comment:
This change is missing from the KIP.

Reply:
Actually let's discuss about that.
Why `super` here?

Reply:
It's from KIP-100 https://cwiki.apache.org/confluence/display/KAFKA/KIP-100+-+Relax+Type+constraints+in+Kafka+Streams+API

Reply:
Thanks, @bbejeck !

@joan38 hmm. I'm not sure why it wouldn't work. All it's saying is that we want:
* a Serde for exactly K
* a Serde for exactly V
* a partitioner (function) that takes 1: anything that can be assigned to K and 2: anything that can be assigned to V

This seems like a perfectly reasonable set of constraints, and it accurately reflects our actual constraints:
* when we're serializing/deserializing, we need to have the exact right serde for the type in question, otherwise, we'll lose some information when we round-trip
* when we're picking a partition, we don't need exactly K and V, but we are processing a <K,V> document, so we need to be able to pass (k:K,v:V) arguments to the partition function. But if you want to write a more generic partition function that, say, picks partitions for any `Object` based on its `hashCode`, then this is fine by us.

I bet that there's some tricky type-bound scala syntax that will make it work. If you run out of ideas, I can try pulling the PR down and banging my head against it, too.

Reply:
@vvcephei Let me try that again tonight. I keep you updated

Reply:
Alright it just worked, so I think I was tired when I did it before

Reply:
:) an underappreciated component in programming